windows/src/desktop/kmshell/install/Keyman.Configuration.UI.InstallFile.pas

```diff
@@ -0,0 +1,274 @@
+unit Keyman.Configuration.UI.InstallFile;
+
+interface
+
+uses
+  System.Classes,
+
+  keymanapi_tlb;
+
+type
+  TInstallFile = class sealed
+  private
+    class procedure AddDefaultLanguageHotkey(
+      Keyboard: IKeymanKeyboardInstalled); static;
+    class procedure AddDefaultLanguageHotkeys(
+      InstalledKeyboards: array of IKeymanKeyboardInstalled); static;
+    class procedure InstallKeyboardPackageLanguage(
+      FPackage: IKeymanPackageInstalled; const BCP47: string); static;
+  public
+    class function BrowseAndInstallKeyboardFromFile(Owner: TComponent): Boolean; static;
+    class function Execute(KeyboardFileNames: TStrings; const FirstKeyboardFileName: string; FSilent, FNoWelcome: Boolean; const LogFile: string): Boolean; overload; static;
+    class function Execute(Owner: TComponent; const FileName: string; ASilent, ANoWelcome: Boolean; const LogFile, BCP47: string): Boolean; overload; static;
+    class function Execute(Owner: TComponent; const FileNames: TStrings; ASilent: Boolean): Boolean; overload; static;
+  end;
+
+implementation
+
+uses
+  System.SysUtils,
+  System.Win.ComObj,
+  Vcl.Controls,
+  Vcl.Dialogs,
+
+  Keyman.Configuration.UI.KeymanProtocolHandler,
+  Keyman.Configuration.UI.MitigationForWin10_1803,
+  kmint,
+  UfrmHTML,
+  UfrmInstallKeyboard,
+  UtilWaitForTSF;
+
+class function TInstallFile.Execute(KeyboardFileNames: TStrings; const FirstKeyboardFileName: string; FSilent, FNoWelcome: Boolean;
```

Comment:
Is this method new?  I can tell that the rest of this file is a refactor of `UfrmInstallKeyboard.pas`, but I see no origin for this method and am having trouble validating it aas a result.

Reply:
It's basically code extracted from initprog.pas, and adding the `TKeymanProtocolHandler` calls.